rfcs/0006-merkle-tree/0006-merkle-tree.md

```diff
@@ -0,0 +1,151 @@
+
+CKB uses Complete Binary Merkle Tree(CBMT) to generate `Merkle Root` and `Merkle Proof` for a static list of items. Currently, CBMT is used to calculate `Transactions Root`. Basically, CBMT is a ***complete binary tree***, in which every level, except possibly the last, is completely filled, and all nodes are as far left as possible. And it is also a ***full binary tree***, in which every node other than the leaves has two children. Compare with other Merkle trees, the hash computation of CBMT is minimal, as well as the proof size.
+
+## Node Orginazation
+
+For the sake of illustration, we order the tree nodes from ***top to bottom*** and ***left to right***. In CBMT with `n` items, root is the `first` node, and the first item's hash is the `n-th` node, second is the `n+1-th` node, etc. We chose this orginazation because it is easy to caculate the node order for an item.
+
+For example, CBMT with 6 items(suppose the hashes are `[T0, T1, T2, T3, T4, T5]`) and CBMT with 7 items(suppose the hashes are `[T0, T1, T2, T3, T4, T5, T6]`) is shown below:
+
+```
+        with 6 items                       with 7 items
+
+              B0                                 B0
+             /  \                               /  \
+           /      \                           /      \
+         /          \                       /          \
+       /              \                   /              \
+      B1              B2                 B1              B2
+     /  \            /  \               /  \            /  \
+    /    \          /    \             /    \          /    \
+   /      \        /      \           /      \        /      \  
+  B3      B4      TO      T1         B3      B4      B5      T0
+ /  \    /  \                       /  \    /  \    /  \
+T2  T3  T4  T5                     T1  T2  T3  T4  T5  T6
+```
+
+Specially, the tree with 0 item is empty(0 node) and its root is H256::zero.
+
+## Tree Struct
+
+CBMT can be represented in a very space-efficient way, using an array alone. Nodes in the array are presented in ascending order.
+
+For example, the two trees above can be represented as:
+
+```
+[B0, B1, B2, B3, B4, T0, T1, T2, T3, T4, T5]
+[B0, B1, B2, B3, B4, B5, T0, T1, T2, T3, T4, T5, T6]
+```
+
+Suppose a CBMT with `n` items, the size of the array would be `2n-1`, the index of item i(start at 0) is `i+n-1`. For node at `i`, the index of its parent is `(i-1)/2`, the index of its sibling is `(i+1)^1-1`(`^` is xor) and the indexes of its children are `[2i+1, 2i+2]`.
+
+## Merkle Proof
+
+Merkle Proof can provide a proof for existence of one or more items. Only sibling of the nodes along the path that form leaves to root, excluding the nodes already in the path, should be included in the proof. We also specify that ***the nodes in the proof should presented in descending order***. For example, if we want to show that `[T1, T4]` is in the list of 6 items above, only nodes `[T5, T0, B3]` should be included in the proof.
```

Comment:
> ***the nodes in the proof should presented in descending order***

- `must` is better than `should` 
- `presented` -> `present`

Reply:
should is better, or just use "is".